### PR TITLE
fix: web vitals tracking events

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Using native web-vitals package to enable tracking for extension as well.
- It's also more extensive:
Screenshot 2023-09-28 at 15 29 27

Example in extension:
Screenshot 2023-09-28 at 15 28 51

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

**Comment by idoshamun** [comment]
@rebelchris we should track the extension as well

**Comment by rebelchris** [comment]
@idoshamun Good point, decided to refactor to use native `web-vitals` note they provide way more data, no sure we actually need all of it or if it bloats analytics too much.

**Comment by idoshamun** [comment]
@rebelchris can you share a complete object as an example please?

**Comment by rebelchris** [comment]
@idoshamun 

Example for `TTFB`
```
{
    "event_name": "TTFB",
    "extra": "{\"name\":\"TTFB\",\"value\":2.4000000059604645,\"rating\":\"good\",\"delta\":2.4000000059604645,\"entries\":[{\"name\":\"chrome-extension://dhhaojmcngfjmoinjljlkdknbcildjlg/index.html\",\"entryType\":\"navigation\",\"startTime\":0,\"duration\":212.60000000149012,\"navigationId\":\"ed2d4562-8d9d-4694-b0e9-1724f7da56b0\",\"initiatorType\":\"navigation\",\"deliveryType\":\"\",\"nextHopProtocol\":\"\",\"renderBlockingStatus\":\"non-blocking\",\"contentType\":\"text/html\",\"workerStart\":0,\"redirectStart\":0,\"redirectEnd\":0,\"fetchStart\":2.4000000059604645,\"domainLookupStart\":2.4000000059604645,\"domainLookupEnd\":2.4000000059604645,\"connectStart\":2.4000000059604645,\"secureConnectionStart\":2.4000000059604645,\"connectEnd\":2.4000000059604645,\"requestStart\":2.4000000059604645,\"responseStart\":2.4000000059604645,\"firstInterimResponseStart\":0,\"responseEnd\":20.600000001490116,\"transferSize\":1034,\"encodedBodySize\":734,\"decodedBodySize\":734,\"responseStatus\":200,\"serverTiming\":[],\"unloadEventStart\":0,\"unloadEventEnd\":0,\"domInteractive\":27.5,\"domContentLoadedEventStart\":212.5,\"domContentLoadedEventEnd\":212.5,\"domComplete\":212.5,\"loadEventStart\":212.5,\"loadEventEnd\":212.60000000149012,\"type\":\"navigate\",\"redirectCount\":0,\"activationStart\":0,\"notRestoredReasons\":null,\"systemEntropy\":\"normal\",\"criticalCHRestart\":0}],\"id\":\"v3-1695908244030-8490804431212\",\"navigationType\":\"navigate\"}"
}
```

Example for `LCP`:
```
{
    "event_name": "LCP",
    "extra": "{\"name\":\"LCP\",\"value\":527.1000000014901,\"rating\":\"good\",\"delta\":527.1000000014901,\"entries\":[{\"name\":\"\",\"entryType\":\"largest-contentful-paint\",\"startTime\":527.1000000014901,\"duration\":0,\"navigationId\":\"ed2d4562-8d9d-4694-b0e9-1724f7da56b0\",\"size\":48320,\"renderTime\":527.1,\"loadTime\":501.8,\"firstAnimatedFrameTime\":0,\"id\":\"\",\"url\":\"https://res.cloudinary.com/daily-now/image/upload/f_auto,q_auto/v1/posts/f7a36bca72ff140acc0b9043a860a18e\"}],\"id\":\"v3-1695908244030-3807810342118\",\"navigationType\":\"navigate\"}"
}
```

**Comment by idoshamun** [comment]
@rebelchris it does look pretty big, the entries specifically. Can you look per event and extract the relevant data you need please?

**Comment by rebelchris** [comment]
@idoshamun I think we can simply omit the entries to be honest.
That's what we got from the next one basically.

**Comment by idoshamun** [comment]
@rebelchris ok! we can always add it again if needed. If possible please also add sampling rate (dunno if it's easily possible).
If not we can just generate a random value to test sampling rate. 10% would be good I guess

**Comment by rebelchris** [comment]
@idoshamun See new proposal, pretty lean i would say:
Screenshot 2023-09-28 at 15 44 59

As for sampling, not supported in the package, can we rely on data for it, or should we write custom sampling script?

**Review by capJavert — COMMENTED** [review]
Looking good, love how lean hook is 👌 , as you said just align it with what next offers so we can switch to their hook down the line.

File: packages/shared/src/hooks/useWebVitals.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      const { name, entries, id, ...rest } = metric;
+      trackEvent({ event_name: name, extra: JSON.stringify(rest) });
+    },
+    [trackEvent],
+  );
+
+  // Have to wait for document to load and ensure it's not run on SSR
+  useEffect(() => {
+    // We apply sampling rate of 10%
+    if (Math.random() < 0.1) {
```
Comment (capJavert) [inline packages/shared/src/hooks/useWebVitals.ts:19]: Maybe sampling rate through `env` so we can modify without code changes?

Comment (rebelchris) [reply packages/shared/src/hooks/useWebVitals.ts:19]: I can do, but it's tricky with extension vs webapp in that case.
So not sure it actually adds much.

**Review by capJavert — APPROVED** [review]
Good work. It would be a good idea to maybe setup a similar dashboard to Vercel speed insight one so we can compare at the start at least.

File: packages/shared/src/hooks/useWebVitals.ts (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/hooks/useWebVitals.ts:19]: We do have the option to split inside data analysis between webapp/extension
We can evaluate if it adds anything basically.

Good point though.
Let's try for now and see what data suggests.

**Pull request merged**